web/client/components/mapcontrols/annotations/AnnotationsEditor.jsx

```diff
@@ -764,6 +769,16 @@ class AnnotationsEditor extends React.Component {
         );
     }
 
+    allGeometryIsValid = () => {
+        let allGeometryIsValid = true;
+        if (this.props?.editing?.features && this.props.editing?.features?.length > 0) {
+            this.props.editing.features.map((p) => {
+                if (!p?.properties?.isValidFeature) { allGeometryIsValid = false; }
+            });
+        }
+        return allGeometryIsValid;
+    };
```

Comment:
personally i don't like this too much because the check "this.props.editing?.features?.length > 0" is superfluous and in this case it would have been better to use a .forEach instead of a .map

but i think this one it's shorter and easier to read
```suggestion
const areAllGeometriesValid = this.props?.editing?.features?.reduce((valid, feature) => {
    return valid && p?.properties?.isValidFeature;
}, true);
return areAllGeometriesValid;
```

or better with every makes it even clearer

```suggestion
let areAllGeometriesValid = true;
if (this.props?.editing?.features) {
    areAllGeometriesValid = this.props?.editing?.features?.every(feature => feature?.properties?.isValidFeature)
}
return areAllGeometriesValid ;
```

Reply:
Thanks, used second version of suggested changes but modified condition to check if this.props?.editing?.features is array and removed optional chaining before "every" since we already check array existence in condition.